Description: Port to exo-1
 Patch slightly updated to patch configure.in instead of configure.in.in
 (not available in released tarballs), and to apply cleanly against 0.1.2.
Origin: backport, http://bugzilla.xfce.org/attachment.cgi?id=2882
Bug: http://bugzilla.xfce.org/show_bug.cgi?id=6295
---
 configure.in                    |    3 ++-
 thunar-plugin/audio-tags-page.c |   22 +++++++---------------
 thunar-plugin/tag-renamer.c     |   16 +++-------------
 3 files changed, 12 insertions(+), 29 deletions(-)

diff --git a/configure.in b/configure.in
index ed3099b..c345c5c 100644
--- a/configure.in
+++ b/configure.in
@@ -64,7 +64,8 @@ XDT_I18N([@LINGUAS@])
 dnl ***********************************
 dnl *** Check for required packages ***
 dnl ***********************************
-XDT_CHECK_PACKAGE([EXO], [exo-0.3], [0.3.1.3])
+XDT_CHECK_PACKAGE([GTK], [gtk+-2.0], [2.12.0])
+XDT_CHECK_PACKAGE([EXO], [exo-1], [0.5.0])
 XDT_CHECK_PACKAGE([THUNARX], [thunarx-2], [1.0.1])
 XDT_CHECK_PACKAGE([TAGLIB], [taglib], [1.4])
 
diff --git a/thunar-plugin/audio-tags-page.c b/thunar-plugin/audio-tags-page.c
index 7e5dab5..40b226f 100644
--- a/thunar-plugin/audio-tags-page.c
+++ b/thunar-plugin/audio-tags-page.c
@@ -89,7 +89,6 @@ struct _AudioTagsPage
   ThunarxPropertyPage __parent__;
 
   /* Widgets */
-  GtkTooltips     *tooltips;
   GtkWidget       *table;
   GtkWidget       *save_button;
   GtkWidget       *info_button;
@@ -281,10 +280,6 @@ audio_tags_page_init (AudioTagsPage *page)
 
   gtk_container_set_border_width (GTK_CONTAINER (page), 8);
 
-  /* Allocate the shared tooltips */
-  page->tooltips = gtk_tooltips_new ();
-  exo_gtk_object_ref_sink (GTK_OBJECT (page->tooltips));
-
   /* Main container */
   vbox = gtk_vbox_new (FALSE, 8);
   gtk_container_add (GTK_CONTAINER (page), vbox);
@@ -316,7 +311,7 @@ audio_tags_page_init (AudioTagsPage *page)
   /* Track spin button */
   spin = gtk_spin_button_new (GTK_ADJUSTMENT (adjustment), 1, 0);
   exo_mutual_binding_new (G_OBJECT (adjustment), "value", G_OBJECT (page), "track");
-  gtk_tooltips_set_tip (page->tooltips, spin, _("Enter the track number here."), NULL);
+  gtk_widget_set_tooltip_text (GTK_WIDGET (spin), _("Enter the track number here."));
   gtk_container_add (GTK_CONTAINER (alignment), spin);
   gtk_label_set_mnemonic_widget (GTK_LABEL (label), spin);
   gtk_widget_show (spin);
@@ -339,7 +334,7 @@ audio_tags_page_init (AudioTagsPage *page)
   /* Year spin button */
   spin = gtk_spin_button_new (GTK_ADJUSTMENT (adjustment), 1, 0);
   exo_mutual_binding_new (G_OBJECT (adjustment), "value", G_OBJECT (page), "year");
-  gtk_tooltips_set_tip (page->tooltips, spin, _("Enter the release year here."), NULL);
+  gtk_widget_set_tooltip_text (GTK_WIDGET (spin), _("Enter the release year here."));
   gtk_container_add (GTK_CONTAINER (alignment), spin);
   gtk_label_set_mnemonic_widget (GTK_LABEL (label), spin);
   gtk_widget_show (spin);
@@ -354,7 +349,7 @@ audio_tags_page_init (AudioTagsPage *page)
   /* Artist entry */
   entry = gtk_entry_new ();
   exo_mutual_binding_new (G_OBJECT (entry), "text", G_OBJECT (page), "artist");
-  gtk_tooltips_set_tip (page->tooltips, entry, _("Enter the name of the artist or author of this file here."), NULL);
+  gtk_widget_set_tooltip_text (GTK_WIDGET (entry), _("Enter the name of the artist or author of this file here."));
   gtk_table_attach (GTK_TABLE (page->table), entry, 1, 4, 1, 2, GTK_EXPAND | GTK_FILL, 0, 0, 0);
   gtk_label_set_mnemonic_widget (GTK_LABEL (label), entry);
   gtk_widget_show (entry);
@@ -369,7 +364,7 @@ audio_tags_page_init (AudioTagsPage *page)
   /* Title entry */
   entry = gtk_entry_new ();
   exo_mutual_binding_new (G_OBJECT (entry), "text", G_OBJECT (page), "title");
-  gtk_tooltips_set_tip (page->tooltips, entry, _("Enter the song title here."), NULL);
+  gtk_widget_set_tooltip_text (GTK_WIDGET (entry), _("Enter the song title here."));
   gtk_table_attach (GTK_TABLE (page->table), entry, 1, 4, 2, 3, GTK_EXPAND | GTK_FILL, 0, 0, 0);
   gtk_widget_show (entry);
 
@@ -383,7 +378,7 @@ audio_tags_page_init (AudioTagsPage *page)
   /* Album entry */
   entry = gtk_entry_new ();
   exo_mutual_binding_new (G_OBJECT (entry), "text", G_OBJECT (page), "album");
-  gtk_tooltips_set_tip (page->tooltips, entry, _("Enter the album/record title here."), NULL);
+  gtk_widget_set_tooltip_text (GTK_WIDGET (entry), _("Enter the album/record title here."));
   gtk_table_attach (GTK_TABLE (page->table), entry, 1, 4, 3, 4, GTK_EXPAND | GTK_FILL, 0, 0, 0);
   gtk_widget_show (entry);
   
@@ -397,7 +392,7 @@ audio_tags_page_init (AudioTagsPage *page)
   /* Comment entry */
   entry = gtk_entry_new ();
   exo_mutual_binding_new (G_OBJECT (entry), "text", G_OBJECT (page), "comment");
-  gtk_tooltips_set_tip (page->tooltips, entry, _("Enter your comments here."), NULL);
+  gtk_widget_set_tooltip_text (GTK_WIDGET (entry), _("Enter your comments here."));
   gtk_table_attach (GTK_TABLE (page->table), entry, 1, 4, 4, 5, GTK_EXPAND | GTK_FILL, 0, 0, 0);
   gtk_widget_show (entry);
 
@@ -411,7 +406,7 @@ audio_tags_page_init (AudioTagsPage *page)
   /* Genre combo box */
   combo = gtk_combo_box_entry_new_text ();
   exo_mutual_binding_new (G_OBJECT (GTK_BIN (combo)->child), "text", G_OBJECT (page), "genre");
-  gtk_tooltips_set_tip (page->tooltips, combo, _("Select or enter the genre of this song here."), NULL);
+  gtk_widget_set_tooltip_text (GTK_WIDGET (combo), _("Select or enter the genre of this song here."));
   gtk_table_attach (GTK_TABLE (page->table), combo, 1, 4, 5, 6, GTK_FILL, 0, 0, 0);
   gtk_widget_show (combo);
 
@@ -511,9 +506,6 @@ audio_tags_page_finalize (GObject *object)
   if (G_LIKELY (page->genre != NULL))
     g_free (page->genre);
 
-  /* Release the tooltips */
-  g_object_unref (G_OBJECT (page->tooltips));
-
   (*G_OBJECT_CLASS (audio_tags_page_parent_class)->finalize) (object);
 }
 
diff --git a/thunar-plugin/tag-renamer.c b/thunar-plugin/tag-renamer.c
index acb694e..bdbb17c 100644
--- a/thunar-plugin/tag-renamer.c
+++ b/thunar-plugin/tag-renamer.c
@@ -116,9 +116,6 @@ struct _TagRenamer
 {
   ThunarxRenamer __parent__;
 
-  /* Widgets */
-  GtkTooltips     *tooltips;
-
   /* Properties */
   TagRenamerFormat format;
   gboolean         replace_spaces;
@@ -209,10 +206,6 @@ tag_renamer_init (TagRenamer *tag_renamer)
   GEnumClass     *klass;
   gint            n;
 
-  /* Allocate shared tooltips */
-  tag_renamer->tooltips = gtk_tooltips_new ();
-  exo_gtk_object_ref_sink (GTK_OBJECT (tag_renamer->tooltips));
-
   table = gtk_table_new (2, 3, FALSE);
   gtk_table_set_row_spacings (GTK_TABLE (table), 6);
   gtk_table_set_col_spacings (GTK_TABLE (table), 12);
@@ -245,14 +238,14 @@ tag_renamer_init (TagRenamer *tag_renamer)
 
   button = gtk_check_button_new_with_mnemonic (_("_Underscores"));
   exo_mutual_binding_new (G_OBJECT (button), "active", G_OBJECT (tag_renamer), "replace-spaces");
-  gtk_tooltips_set_tip (tag_renamer->tooltips, button, _("Activating this option will replace all spaces in the target filename "
-        "with underscores."), NULL);
+  gtk_widget_set_tooltip_text (GTK_WIDGET (button), _("Activating this option will replace all spaces in the target filename "
+        "with underscores."));
   gtk_table_attach (GTK_TABLE (table), button, 2, 3, 0, 1, GTK_FILL, 0, 0, 0);
   gtk_widget_show (button);
 
   button = gtk_check_button_new_with_mnemonic (_("_Lowercase"));
   exo_mutual_binding_new (G_OBJECT (button), "active", G_OBJECT (tag_renamer), "lowercase");
-  gtk_tooltips_set_tip (tag_renamer->tooltips, button, _("If you activate this, the resulting filename will only contain lowercase letters."), NULL);
+  gtk_widget_set_tooltip_text (GTK_WIDGET (button), _("If you activate this, the resulting filename will only contain lowercase letters."));
   gtk_table_attach (GTK_TABLE (table), button, 2, 3, 1, 2, GTK_FILL, 0, 0, 0);
   gtk_widget_show (button);
 
@@ -272,9 +265,6 @@ tag_renamer_finalize (GObject *object)
 {
   TagRenamer *tag_renamer = TAG_RENAMER (object);
 
-  /* release the tooltips */
-  g_object_unref (G_OBJECT (tag_renamer->tooltips));
-
   /* Free strings */
   g_free (tag_renamer->artist);
   g_free (tag_renamer->title);

